src/clarity_vm/database/marf.rs

```diff
@@ -826,11 +546,11 @@ impl<'a> ClarityBackingStore for WritableMarfStore<'a> {
             })
     }
 
-    fn get_with_proof(&mut self, key: &str) -> Option<(String, TrieMerkleProof<StacksBlockId>)> {
+    fn get_with_proof(&mut self, key: &str) -> Option<(String, String)> {
```

Comment:
Why is this changed to a `String`?

Reply:
This change was made in order to remove an instance of the `vm`'s dependency on `chainstate`, and by extension, `rusqlite`.

```Rust
// chainstate/stacks/index/proofs.rs
pub struct TrieMerkleProof<T: MarfTrieId>(pub Vec<TrieMerkleProofType<T>>);

// chainstate/stacks/index/mod.rs
pub trait MarfTrieId:
    PartialEq
    + Clone
    + std::fmt::Display
    + std::fmt::Debug
    + rusqlite::types::ToSql
    + rusqlite::types::FromSql
    + std::convert::From<[u8; 32]>
    + std::convert::From<MARFValue>
    + ::net::StacksMessageCodec
{
...
}
```

Changing the return type of the `get_with_proof` function was a simple way to get rid of these dependencies. Another solution would be to put the `ToSql` and `FromSql` trait inheritance behind a feature flag, and hoist the `MarfTrieId` type out of `chainstate` and into the new `types` module.

Reply:
Why is it a problem that the `clarity_vm` crate depend on `rusqlite`?  Doesn't it do so already?  Can't you just hoist `MarfTrieId` into the `types` module?

Reply:
I think we can worry about the structured merkle proof response when we actually need them -- for now, the only use case of this method immediately serializes the proof (and this change doesn't actually destroy the structure of the response: it could be deserialized). Once we're implementing things that actually want these proofs, we'll have a much better idea about how to change the response (does an associated type make sense here for example? a non-marf implementation of the clarity db, for example, probably would want to return a totally different type).

Reply:
@jcnelson I went ahead and implemented this as it wasn't that much of an effort. @kantai given that I've made this change, do you still have a preference either way?

Reply:
Yep, that works for me